### PR TITLE
add mongodb

### PR DESCRIPTION
**Pull request by YoungFr**
Updated the relevant dependencies and LICENSE.
Open source mongodb to GitHub trpc ecosystem.

### CODE REVIEW

**Comment by YoungFr** [comment]
I have read the CLA Document and I hereby sign the CLA

File: LICENSE
Comment (liuzengh) [inline LICENSE:1]: add license for package appearing in go.mod, eg: " mongo-driver"

MIT: goconvey, gomonkey/v2, testify

Apache-2.0： mock, mongo-driver, yaml.v3

File: mongodb/curd.go [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// MongoArgFilter mongo operation instruction
+const (
+	MongoArgFilter               = "filter"
```
Comment (liuzengh) [inline mongodb/curd.go:17]: These constants do not need to be exported

File: mongodb/codec.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	"trpc.group/trpc-go/trpc-go/codec"
+)
+
+func init() {
+	codec.Register("mongodb", nil, DefaultClientCodec)
+}
+
+// default codec
+var (
+	DefaultClientCodec = &ClientCodec{}
```
Comment (liuzengh) [inline mongodb/codec.go:17]: do not need to be exported

File: mongodb/plugin.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+type Config struct {
+	MinOpen        uint64        `yaml:"min_open"`        // Minimum number of simultaneous online connections
+	MaxOpen        uint64        `yaml:"max_open"`        // The maximum number of simultaneous online connections
+	MaxIdleTime    time.Duration `yaml:"max_idle_time"`   // Maximum idle time per link
+	ReadPreference string        `yaml:"read_preference"` // reference on read
+}
+
+// MongoPlugin is used for plug-in default initialization,
+// used to load mongo proxy connection parameter configuration.
+type MongoPlugin struct{}
```
Comment (liuzengh) [inline mongodb/plugin.go:32]: do not need to be exported

File: LICENSE
Comment (YoungFr) [reply LICENSE:1]: add mongo-driver in LICENSE

File: mongodb/curd.go (hunk 1 above)
Comment (YoungFr) [reply mongodb/curd.go]: modify these constants to be non-exported

File: mongodb/codec.go (hunk 2 above)
Comment (YoungFr) [reply mongodb/codec.go]: fixed

File: mongodb/plugin.go (hunk 3 above)
Comment (YoungFr) [reply mongodb/plugin.go]: fixed

**Comment by sandyskies** [comment]
Please add some PR description

**Comment by YoungFr** [comment]
> Please add some PR description

done

**Review by liuzengh — APPROVED** [review]
LGTM

**Pull request merged**